lib/eclipse/EclipseState/Schedule/MSW/Compsegs.cpp

```diff
@@ -168,18 +168,61 @@ namespace Opm {
             }
 
             compseg.m_segment_number = segment_number;
+
+            // when depth is default or zero, we obtain the depth of the completion based on the information
+            // of the related segments
             if (compseg.m_center_depth == 0.) {
-                // using the depth of the segment node as the depth of the completion
-                // TODO: now only one completion for one segment is hanlded,
-                // TODO: later we will try to handle more than one completion for each segment,
-                // TODO: which will be a linear interpolation based on the segment node depth
-                // TODO: in the same branch, while the actually way is not clear yet
-                const int segment_location = segment_set.numberToLocation(segment_number);
-                compseg.m_center_depth = segment_set[segment_location].depth();
+                compseg.calculateCenterDepthWithSegments(segment_set);
             }
         }
     }
 
+    void Compsegs::calculateCenterDepthWithSegments(const SegmentSet& segment_set) {
+
+        // the depth and distance of the segment to the well head
+        const int segment_location = segment_set.numberToLocation(m_segment_number);
+        const double segment_depth = segment_set[segment_location].depth();
+        const double segment_distance = segment_set[segment_location].totalLength();
+
+        // for top segment or the distance of segment and perforation are the same
+        // no interpolation is needed
+        const double center_distance = (m_distance_start + m_distance_end) / 2.0;
+        if (center_distance == segment_distance || m_segment_number == 1) {
```

Comment:
I would assume that `m_segment_number == 1` would be enough to check if we are actually considering the top segment? Iff the distance test is indeed required it needs some "epsilon protection" - checking two `double` values for equality with `==` is probably not correct?

Reply:
Yes. The "m_segment_number == 1" is for top segment, the other one is consider the case that the distance is input as exactly the same with the segment distance.  If they are slightly different, then we go to interpolation. We can remove `center_distance == segment_distance`, let it always do the interpolation, which might kill slightly more computation, which we might not care.  How do you think?

Reply:
I decided to remove the comparison. It will increase the computation a little bit, while we do not worry about that.

Reply:
> [...] which might kill slightly more computation, which we might not care. How do you think?

Always interpolate - clarity is paramount for this code.